src/routes/github/create-branch/github-create-branch-get.ts

```diff
@@ -31,7 +37,10 @@ export const GithubCreateBranchGet = async (req: Request, res: Response, next: N
 	if (!issueKey) {
 		return next(new Error(Errors.MISSING_ISSUE_KEY));
 	}
-	const subscriptions = await Subscription.getAllForHost(jiraHost, gitHubAppConfig.gitHubAppId || null);
+
+	//TODO  ARC-1823: fix potential subscript get all for host bug
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	const subscriptions = await Subscription.getAllForHost(jiraHost, gitHubAppConfig.gitHubAppId as any || null);
```

Comment:
I need to review the intention of this `Subscriptioin.getAllForHost` when the `gitHubAppId` is undefined. Should it include `gitHubAppId is null` in the where clause? Here it does, but many other places it doesn't. Quite inconsistent.

Reply:
And possibly raise another PR.

Reply:
yes, that's the point.  If no id is specified, I needs to get the ones specifically with no id in the database (ie `null`).  If undefined is used, it gets all subscriptions as the where clause wouldn't be added.

Reply:
Okay, got it. Although I still feel using `null` vs `undefined` to distinguish behaviour is unfamiliar. And could be dangerous when it hooked up to some deleting routes.